File: nodejs/7 kyu/107.js
// https://www.codewars.com/kata/56044de2aa75e28875000017/train/javascript

// Passed

function compoundArray(a, b) {
  let maxLength = Math.max(a.length, b.length);
  let compoundedArray = [];
  for (let idx = 0; idx < maxLength; idx++) {
    if (idx < a.length) compoundedArray.push(a[idx]);
    if (idx < b.length) compoundedArray.push(b[idx]);
  }

  return compoundedArray;
}

const output = compoundArray([11, 12], [21, 22, 23, 24]);
console.log(output);

File: python/7 kyu/107.py
# https://www.codewars.com/kata/58fd91af13b0012e8e000010/train/python

# Passed

from collections import defaultdict

def strange_coach(players):
    letter_count_dict = defaultdict(int)
    for player in players:
        letter_count_dict[player[0]] += 1

    valid_players = [player for player in players if letter_count_dict[player[0]] >= 5]
    if not valid_players:
        return "forfeit"

    first_letters = [letter for letter in letter_count_dict if letter_count_dict[letter] >= 5]
    first_letters.sort()

    result = "".join(first_letters)
    return result

output = strange_coach(["babic","keksic","boric","bukic",
                        "sarmic","balic","kruzic","hrenovkic",
                        "beslic","boksic","krafnic","pecivic",
                        "klavirkovic","kukumaric","sunkic","kolacic",
                        "kovacic","prijestolonasljednikovic"])
print(output)